problem/0002-isolated-integration-tests.md

```diff
@@ -0,0 +1,151 @@
+
+The Linkerd integration test suite is extensive and effective at catching
+bugs, but it does not provide a very positive UX experience to new or existing
+Linkerd developers.
+
+It is also unable to provide many options for testing complex cluster
+scenarios such as multi-cluster installations and cluster failure scenarios.
+
+The are several reasons for these existing problems and each should be
+addressed by the change this RFC is proposing.
+
+### UX
+
+Running specific tests is not currently possible without knowing how the
+internal testing scripts work. This can be better so designed so that
+developers are able to test specific Linkerd integrations without running the
+entire test suite, or looking at the script internals.
+
+After this change, the process for running specific integration tests will be
+more perceptible to developers.
+
+### Constraints
+
+The integration test setup checks require that certain conditions are
+satisfied by the given cluster. A surprising condition is that no pre-existing
+Linkerd installation resource may exist; if it does then it is deleted. These
+conditions exist because the test suite is constrained to a single
+cluster--running each test in its own namespace.
+
+After this change, the test suite should isolate tests by cluster. This will
+allow for each test to require any pre-existing state that is required without
+affecting later runs or existing installations.
+
+Another possibility with less constraints on the cluster state is parallel
+execution of integration tests.
+
+### Extensibility
+
+Extending integration tests for more complex installation setups is not
+accessible.
+
+After this change, the testing process should be extendable in order to test
+new features being added to Linkerd. This can include things like checking the
+behavior of multi-cluster installations as well as the handling of errors in
+cluster failure scenarios.
+
+# Design proposal (Step 2)
+
+[design-proposal]: #design-proposal
+
+**Note**: This should be completed as part of `Step 2`.
+
+This is the technical portion of the RFC. Explain the design in sufficient
+detail that:
+
+- Its interaction with other features is clear
+- It is reasonably clear how the contribution would be implemented
+- Corner cases are dissected by example
+- Dependencies on libraries, tools, projects or work that isn't yet complete
+- Use Cases
+- Goals
+- Non-Goals
+- Deliverables
+
+# Prior art
+
+[prior-art]: #prior-art
+
+### KinD
+
+Kubernetes isolates it's testing through
+[KinD](https://github.com/kubernetes-sigs/kind) which Linkerd already relies
+on in several scripts.
+
+Isolating through KinD clusters will provide stronger isolation than namespace
+because creating a new a cluster for each test ensures previous runs do not
+affect the current installation.
+
+It may not be possible to always rely on KinD if it is not installed, which is
+why namespace isolation should still be available.
+
+### k3d (and k3s)
```

Comment:
The other big benefit to k3d, imo, is that its new version offers an SDK so that tests could integrate directly without having to require the system has the correct version of k3d installed.